tests/test_estimate_pz_task.py

```diff
@@ -0,0 +1,239 @@
+            )
+            to_delete.append(model_file_knn_lsst)
+        pz_model_knn_lsst = PZModel.read(model_file_knn_lsst)
+
+        task_config_knn = EstimatePZTask.ConfigClass()
+        task_config_knn.pz_algo.retarget(EstimatePZKNNTask)
+        task_config_knn.pz_algo.stage_name = "knn"
+        task_config_knn.pz_algo.output_mode = "return"
+
+        task_knn = EstimatePZTask(True, config=task_config_knn)
+        dd = butler.getDeferred("objectTable", skymap="DC2", tract=3829, patch=1)
+
+        out_knn = task_knn.run(pz_model_knn_lsst, dd)
+        out_knn.pzEnsemble.write_to("output_knn_lsst.hdf5")
+        to_delete.append("output_knn_lsst.hdf5")
+        test_out = qp.read("output_knn_lsst.hdf5")
+        assert isinstance(test_out, qp.Ensemble)
+        assert test_out.npdf == 29358
+        for fdel_ in to_delete:
+            os.unlink(fdel_)
+
+    @unittest.skipIf(CI_HSC_GEN3_DIR is None, "CI_HSC_GEN3 not installed")
+    def test_pz_tasks_ci_hsc(self):
+        model_file_knn_hsc = "model_inform_knn_hsc_wrap.pickle"
+        to_delete = []
+        if not os.path.exists(model_file_knn_hsc):
+            os.system(
+                f"curl -O https://portal.nersc.gov/cfs/lsst/PZ/pz_models/{model_file_knn_hsc}"
+            )
+            to_delete.append(model_file_knn_hsc)
+
+        butler = self.makeButler_ci_hsc(writeable=True)
+        butler.registry.registerDatasetType(self.pzModel_trainz_datasetType)
+        butler.registry.registerDatasetType(self.pzModel_knn_datasetType)
+        butler.registry.registerRun("u/testing/pz_models")
+
+        pzModel_trainz_datasetRef = DatasetRef(
+            self.pzModel_trainz_datasetType,
+            DataCoordinate.from_full_values(
+                self.pzModel_dimension_group,
+                ("HSC",),
+            ),
+            run="u/testing/pz_models",
+        )
+
+        pzModel_knn_datasetRef = DatasetRef(
+            self.pzModel_knn_datasetType,
+            DataCoordinate.from_full_values(
+                self.pzModel_dimension_group,
+                ("HSC",),
+            ),
+            run="u/testing/pz_models",
+        )
+
+        butler.ingest(
+            FileDataset(
+                os.path.join(TEST_DIR, "model_inform_train_z_wrap.pickle"),
+                pzModel_trainz_datasetRef,
+            ),
+        )
+
+        butler.ingest(
+            FileDataset(
+                os.path.join(TEST_DIR, "..", "model_inform_knn_hsc_wrap.pickle"),
+                pzModel_knn_datasetRef,
+            ),
+        )
+
+        subprocess.run(
+            [
+                "pipetask",
+                "run",
+                "--register-dataset-types",
+                "-b",
+                os.path.join(CI_HSC_GEN3_DIR, "DATA"),
+                "-i",
+                "HSC/runs/ci_hsc,u/testing/pz_models",
+                "-o",
+                "u/testing/pz_rail_testing",
```

Comment:
Another problem I foresee here is if you keep re-running this test, it will add more and more runs to this collection, which can eventually become slow to modify. Consider adding some random string here and also cleaning up the collection after the test is finished (with some combination of `butler.pruneDatasets` and `butler.removeRuns`).

Reply:
`tests/cleanup_script` can be used to do the cleanup.   I prefer this b/c rather than automoatically cleaning things up, it lets you investigate what happens which things fail.   At least for the ci_hsc_gen based tests I'd rather stick with this.

Reply:
I still think you should do cleanup on test success. I didn't find an automated way of doing this in unittest. You could do something like:
- `self.ci_hsc_result = 0` in `setUp`
- `self.ci_hsc_result = subprocess.run(` above (instead of just `result = ...`)
- `if self.ci_hsc_result == 0:  # run cleanup in `tearDown`
I'm ambivalent about whether the cleanup commands should be done with butler calls or left as a shell script (but then it should probably be `cleanup.sh`).

Reply:
I just run the cleanup script at the end of the test, since you only get there on success.